From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 30 Dec 2011 17:48:49 +0100
Subject: [PATCH] Revert "clockevents: Set noop handler in
 clockevents_exchange_device()"

This reverts commit aeed6baa702a285cf03b7dc4182ffc1a7f4e4ed6.
It caused some systems to hang on resume (e.g. #653398).
---
 kernel/time/clockevents.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/kernel/time/clockevents.c b/kernel/time/clockevents.c
index 13dfaab..e4c699d 100644
--- a/kernel/time/clockevents.c
+++ b/kernel/time/clockevents.c
@@ -286,7 +286,6 @@ void clockevents_exchange_device(struct clock_event_device *old,
 	 * released list and do a notify add later.
 	 */
 	if (old) {
-		old->event_handler = clockevents_handle_noop;
 		clockevents_set_mode(old, CLOCK_EVT_MODE_UNUSED);
 		list_del(&old->list);
 		list_add(&old->list, &clockevents_released);
-- 
1.7.7.3

